src/js/components/List/List.js

```diff
@@ -357,13 +366,19 @@ const List = React.forwardRef(
                       // accessibility guidelines that focus remains on `ul`
                       listRef.current.focus();
                     },
-                    onMouseOver: () => setActive(index),
-                    onMouseOut: () => setActive(undefined),
+                    onMouseOver: () => {
+                      setLastActive(undefined);
+                      setActive(index);
+                    },
+                    onMouseOut: () => {
+                      setActive(lastActive);
+                    },
                     onFocus: () => {
                       setActive(index);
                       setItemFocus(true);
                     },
                     onBlur: () => {
+                      setLastActive(active);
```

Comment:
I would have expected `onFocus` to set both `active` and `lastActive` and `onBlur` to just clear `active`. Why are we setting last active at `onBlur` time?

Reply:
I agree with this idea, moving the `lastActive` to `onFocus` makes more sense here